### ISSUE TITLE
1.0.0-alpha

### ISSUE THREAD

**Issue by MiloszKrajewski** [MiloszKrajewski/K4os.Compression.LZ4#5]
- [ ] block fast encoder
- [ ] block high encoder
- [ ] block decoder
- [ ] stream encoder
- [ ] stream decoder
- [ ] frame decoder
- [ ] frame encoder
- [ ] xxhash in header
- [ ] friendly API

**Comment by Krakean** [comment]
>Note: lz4 1.8.2 introduces important bugfix but also breaking changes

You mean released 1.8.1.2 or 1.8.2 that one which is in dev branch? Just to be sure :)
And, if we speak about 1.8.2-dev, what important bugfix it made?

Also, I suppose you want 1.0.0-alpha be compatible with 1.8.1(.2?) that you initially picked up, but later anyway plan to upgrade to later version of lz4 with those breaking changes (imho there is nothing wrong in breaking changes, user always can stay with old/previously released version)?

**Comment by MiloszKrajewski** [comment]
1.8.2 fixes some low probability/high impact security issue. It won't happen on PC but might happen on IoT devices. So, I'm too concerned but still it is worth to port it, as once ported it freezes for a while.
It is not a interface breaking change (so no problem for you), but produced output is binary identical so lots of my unit tests will start to fail.

**Comment by Krakean** [comment]
@MiloszKrajewski got it. :) then, if there is no problem for PC, so no reason to rush then with 1.8.2 update :)

**Comment by Krakean** [comment]
@MiloszKrajewski development stalled? :-\

**Comment by MiloszKrajewski** [comment]
Yup. I had a bug and a mental block for a while. And then, after returning from holiday, found and fixed problem in one day. 
Anyway, last thing to do is: more tests (stressing it a little bit) and friendlier API.

**Comment by Krakean** [comment]
@MiloszKrajewski good to see you back.
By the way, any plans to update fresh version of lz4? :) As I assume your source code is based on lz4 1.8.1 (deprecated one?), and now 1.8.2 is available

**Comment by MiloszKrajewski** [comment]
I just release to nuget and wrote some docs on main page. I'll wait for some API feedback and try to port next version. As long as it is just bug fixes it should be relatively easy, but if there are breaking changes it may take while.

**Comment by Krakean** [comment]
@MiloszKrajewski Well, since major version hasn't been changed, I guess its just minor fixes + optimizations, so I guess it shouldn't take much time :)
Plus, would be nice if you will add to readme.md some benchmarks, just have some speed comparing to C's lz4.

**Comment by Krakean** [comment]
@MiloszKrajewski 
>try to port next version

Curious, any progress on this?

**Comment by Krakean** [comment]
@MiloszKrajewski ping :)

**Comment by Krakean** [comment]
@MiloszKrajewski Hello Milosz, when you plan to continue development on this library? Or its no longer maintained?

**Comment by MiloszKrajewski** [comment]
Yes, I do. I'm just doing some online courses now and all my evenings are consumed by writing Scala. I'll come back to this project soon.
Do you have any urgent bugs/issues?

**Comment by Krakean** [comment]
@MiloszKrajewski 
> I'll come back to this project soon.

Glad to hear :)

>Do you have any urgent bugs/issues?

Nah, not at the moment at least :)
Just worry about this project future, its the only C# native decent compression library which is suitable for games. Cant wait to see update sources to latest lz4 master with optimized decompression :) Block/content checksum also would be nice to see

**Comment by Krakean** [comment]
@MiloszKrajewski 
Hello Milosz
No update to lz4 v1.9.1 in near feature? :-\ It has decent decomp speed optimizations which are really nice to have.

**Comment by MiloszKrajewski** [comment]
I pulled 1.9.1 and compared it with 1.8 and change are quite drastic so no tweaking, it would require full port again. On the radar, but functional changes (like true async/await) would take precedence I think.

**Comment by Krakean** [comment]
@MiloszKrajewski 
>I can try latest 1.8.x though, but I'm not sure if this makes sense.

Its better than nothing at all :)

>but functional changes (like true async/await) would take precedence I think.

Oh, good.
Any approximate estimates on new version? :)

**Comment by Krakean** [comment]
@MiloszKrajewski 
Hello Milosz
Any progress on v1.9.1? As I can see by existing of the branch '191' you decided to integrate 1.9.1 after all, right?

**Comment by MiloszKrajewski** [comment]
++version

**Comment by kylemc** [comment]
I see 'add true async read/write interface' in vNext above, but I also wrote up #34 to explain the value to me. Without a fix, I'll have to look for other fast compression options.
